app/validators/well_positions_validator.rb

```diff
@@ -25,8 +25,13 @@ def valid_positions
   def validate(record)
     well_positions = filtered(record.wells).collect(&:position)
 
-    return if (well_positions - valid_positions).empty?
+    invalid_positions = (well_positions - valid_positions)
 
-    record.errors.add(:wells, "must be in positions #{valid_positions.join(',')}")
+    return if invalid_positions.empty?
+
+    invalid_wells = invalid_positions.join(',')
```

Comment:
Do we need to create new variables here? Could we just do the join in the add as the variables are not used anywhere else?